diceware/custom_config.go

```diff
@@ -0,0 +1,74 @@
+package diceware
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/cheggaaa/pb/v3"
+)
+
+type CustomConfig struct {
+	Lang   bool
+	Add    bool
+	Source string
+	Name   string
+}
+
+func (c *CustomConfig) Configure() error {
+	if c.Add && c.Lang {
+		return c.newLanguage()
+	}
+
+	return nil
+}
+
+func (c *CustomConfig) newLanguage() error {
+	if c.Source == "" || c.Name == "" {
+		return fmt.Errorf("Please provide both dictionary source file and language name (--source, --name)")
+	}
+
+	file, err := os.Open(c.Source)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return err
+	}
+
+	dicewarePath := home + "/.diceware/diceware_words_" + c.Name
```

Comment:
What happens if the user passes an invalid name? Eg. (Windows): `pt:BR`

Reply:
now it will fail if it's not valid 👀